### ISSUE TITLE
rustc does not recognize "is-like-msvc" properly?

### ISSUE THREAD

**Issue by pravic** [rust-lang/rust#32818]
[Here](https://gist.github.com/pravic/9d8a5ec285efa187000cb353a9ad3256#file-i786-pc-windows-msvc-json) is a gist with the target specification.

When `"linker": "link.exe"` is specified, I get the following:

```
rustc --target=i786-pc-windows-msvc.json --crate-type=dylib lib.rs
note: LINK : warning LNK4044: unrecognized option '/L'; ignored
LINK : warning LNK4044: unrecognized option '/o'; ignored
LINK : warning LNK4044: unrecognized option '/nodefaultlibs'; ignored
LINK : warning LNK4044: unrecognized option '/L'; ignored
LINK : warning LNK4044: unrecognized option '/Wl,-Bstatic'; ignored
LINK : warning LNK4044: unrecognized option '/Wl,-Bdynamic'; ignored
LINK : warning LNK4044: unrecognized option '/shared'; ignored
```

If omit it, rustc calls `cc`:
```
rustc --target=i786-pc-windows-msvc.json --crate-type=dylib lib.rs
error: could not exec the linker `cc`
```

Why? [This code](https://github.com/rust-lang/rust/blob/master/src/librustc_trans/back/link.rs#L636-L647) shows that `is-like-msvc` must handle it properly. And such things as "-Wl,-Bstatic" is coming from [GnuLinker](https://github.com/rust-lang/rust/blob/master/src/librustc_trans/back/linker.rs#L192-L200).

Meta:

> rustc 1.9.0-nightly (bf5da36f1 2016-04-06)
binary: rustc
commit-hash: bf5da36f1dfae45941ec39ef67a41fdbd22c1a50
commit-date: 2016-04-06
host: i686-pc-windows-msvc
release: 1.9.0-nightly

**Comment by Aatch** [comment]
That option isn't settable from JSON. This https://github.com/rust-lang/rust/blob/master/src/librustc_back/target/mod.rs#L419, plus a few other keys are supported.

It probably should be settable from the JSON.

**Comment by pravic** [comment]
Thanks.

Guess reasonable keys which are missing are

> cpu: "generic".to_string(),
is_like_android: false,
is_like_msvc: false,
is_like_solaris: false,

**Comment by cardoe** [comment]
> Guess reasonable keys which are missing are
>> is_like_android: false,
>> is_like_msvc: false,
>> is_like_solaris: false,

Was there a reason for leaving the android and solaris ones out?

**Comment by pravic** [comment]
Issue was fixed at #32823 and landed with 2016-04-12 nightly.

**Comment by pravic** [comment]
Idk about testcases, may be they needed for this.
